pinot-connectors/pinot-spark-3-connector/src/main/scala/org/apache/pinot/connector/spark/v3/datasource/PinotDataWriter.scala

```diff
@@ -94,6 +115,8 @@ class PinotDataWriter[InternalRow](
     val variables = Map(
       "partitionId" -> partitionId,
       "table" -> tableName,
+      "startTime" -> startTime,
```

Comment:
nit: you can also update javadocs to show the new variables for segment name generation. But do call out that it only works for numeric time columns.